lib/tasks/config.rake

```diff
@@ -102,15 +107,24 @@ namespace :newrelic do
       description
     end
 
-    def format_default_value(spec)
+    def format_default_value(spec, key)
       if spec[:default].is_a?(Proc)
-        '(Dynamic)'
+        return '(Dynamic)' unless default_value_auto?(key)
+        '"auto"'
       else
         "#{spec[:default].inspect}"
       end
     end
 
+    # currently the only 'instrumentation.XXXX' configs that
+    # do not use a default of auto are mongo and excon
+    def default_value_auto?(key)
+      return false if key.match(/mongo/) || key.match(/excon/)
+      key.match(/^instrumentation/)
+    end
```

Comment:
I don't really love the way I'm doing this right now. It seems like it could get accidentally broken pretty easily. 
I'm thinking what might be a better overall solution for this situation is to add another field on the default values in `default_source.rb` to list what we want the docs to use as the default value if it's present.
Something like: 
```
        :'instrumentation.net_http' => {
          :default => instrumentation_value_of(:disable_net_http, :prepend_net_instrumentation),
          :documentation_default => "auto",
          :public => true,
          :type => String,
          :dynamic_name => true,
          :allowed_from_server => false,
          :description => "Controls auto-instrumentation of Net::HTTP at start up.  May be one of [auto|prepend|chain|disabled]."
        },
```

Reply:
I think that's a great idea! I say go for it if you feel motivated or add the snippet to the issue (https://github.com/newrelic/newrelic-ruby-agent/issues/810) if you'd rather wait!

Reply:
Whether it's changed now or later, what do you think about adding tests for these methods?

Reply:
You could use a negative look ahead to combine the 3 separate regex matches into a single one:

```ruby
def default_value_auto?(key)
  key.match?(/^instrumentation(?!.*(?:mongo|excon))/
end
```

Reply:
Updated to not check the config key value instead, and use `documentation_default` field if it exists.